app/controllers/laa_references_controller.rb

```diff
@@ -35,7 +49,11 @@ def resource
   end
 
   def resource_params
-    @resource_params ||= laa_reference_params.select! { |k, _v| k.in? %w[maat_reference defendant_id] }
+    @resource_params ||= if link_attempt_params
```

Comment:
This is confusing so I would start by simplifying this by renaming these things

resource --> laa_reference
resource_params --> link_attempt_params